### ISSUE TITLE
Bug? editor.updateBlock doesn't update the dom node immediately

### ISSUE THREAD

**Issue by man-shar** [TypeCellOS/BlockNote#514]
**Describe the bug**
I'm looking for a prop in my custom block, and if it's not there I'm setting the value by calling editor.updateBlock. But even after calling that, my block's prop is still undefined in the lines below it. Is this intended behaviour? Is editor.updateBlock supposed to be async-ish when it comes to updating the dom?

Weirdly: This seems to work normally in a locally running web app and a deployed website. I am only facing this issue when running this as an app in a docker container.

Even weirder: I'm using Yjs as a backend, and upon page refresh, that prop appears as a data-attribute on the dom node. So it wasn't completely lost. Just something about the update behaviour is weird. Is there anything I can do to force an update to the dom node?

```js
const MyBlock = createReactBlockSpec(
  {
    type: "my-block",
    propSchema: {
      myProp: {
        default: null,
      },
    },
    content: "none",
  },
  {
    render: ({ block, editor }) => {

      if (!block.props.myProp) {
        const updatedBlock = editor.updateBlock(block, {
          props: {
            ...block.props,
            myProp: NEW_RANDOM_PROP_VAL,
          },
        });
        console.log(block.props.myProp); // still undefined!
      }

      return (
          <ReactComponent
            myProp={block.props.myProp}
          />
      );
    },
  }
);
```

Thank you! This project is amazing!

**Misc**
- Node version: 21
- Package manager: npm
- Browser: Chrome

**Comment by man-shar** [comment]
I fixed this by passing in the prop earlier in the chain.
I was running `editor.insertBlock` from a slash menu item. Just passed it in there.

Still don't quite understand why it worked elsewhere (something deployed on cloudflare) and not from a docker container but 🤷 

Tiny bit of debugging I did that might help someone else find the issue:
1. The `BlockContentWrapper` function inside `ReactBlockSpec.tsx` would be called multiple times, sometimes without props and then later (maybe 3rd time) with the desired prop. Not sure why

Will close this issue.
